pkg/glusterblock/controllerserver.go

```diff
@@ -0,0 +1,316 @@
+package glusterblock
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/gluster/gluster-csi-driver/pkg/utils"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/gluster/glusterd2/pkg/api"
+	bapi "github.com/gluster/glusterd2/plugins/blockvolume/api"
+	"github.com/golang/glog"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	defaultVolumeSize int64 = 1000 * utils.MB // default volume size ie 1 GB
+)
+
+// ControllerServer struct of GlusterFS Block CSI driver with supported methods of CSI controller server spec.
+type ControllerServer struct {
+	*GfDriver
+}
+
+// CsiDrvParam stores csi driver specific request parameters.
+// This struct will be used to gather specific fields of CSI driver:
+// For eg. csiDrvName, csiDrvVersion..etc
+// and also gather parameters passed from SC which not part of gluster volcreate api.
+// GlusterCluster - The resturl of gluster cluster
+// GlusterUser - The gluster username who got access to the APIs.
+// GlusterUserToken - The password/token of glusterUser to connect to glusterCluster
+// GlusterVersion - Says the version of the glustercluster running in glusterCluster endpoint.
+// All of these fields are optional and can be used if needed.
+type CsiDrvParam struct {
```

Comment:
if required this also  can be moved  to common place

Reply:
I will do code refactor in separate patch.